base/doc/ltnews38.tex

```diff
@@ -278,67 +278,149 @@ \section{Hooks, sockets and plugs}
-
-As an example to setup a new property which records the current chapter 
-number the following declaration could be used: 
+\section{Document properties and cross-referencing}
+
+Traditional \LaTeX{} uses \cs{label} to record the values of
+two \enquote{local}
+properties of the document: the textual representations of the
+%% FMi added back in, be cause it is the "current", see Lamport:
+\emph{current
+page number} and the
+%%FMi from Lamport
+the \emph{current \textup{\cs{ref}} value} set by
+\cs{refstepcounter} declarations~\cite[p.~209]{38:Lamport}.
+(These declarations are issued by
+sectioning commands, by numbered environments, and by \cs{item} in an
+\env{enumerate} or a similar environment.)
+
+These two recorded values can then be accessed and typeset
+(from anywhere in the next run of the document) by use of the
+(non-expandable) commands \cs{ref} and \cs{pageref}
+%%FMi new
+using the \emph{key} that was assigned to the values with the \cs{label} command.
+This supported basic cross-referencing (within a document) using
+these recorded values to provide both page-related and counter-related
+information (such as the page xvii or the subsection number 4.5.2).\footnote{In the Spring 2023 release of \LaTeX{}, the \cs{label} command was extended
+to record, in addition, both a title (such as the text used in a
+section head) and the \emph{logical name} used
+for an associated link target
+%%FMi new
+if these have been set by packages such as  \pkg{nameref} or \pkg{hyperref}.}
+
+
+%In the Spring 2023 release, the \cs{label} command was extended
+%to record, in addition, both a title (such as the text used in a
+%section head) and the \emph{logical name} used
+%for an associated link target
+%%FMi new
+%if these have been set by packages such as  \pkg{nameref} or \pkg{hyperref}.
+
+%%FMi: meanwhile sounds as if it just happened after 2023 Spring ...
+%%     So moved the above text into a footnote (is anyway old news) and
+%%    did some rewrites below.
+%%
+%Meanwhile, other packages had been extending
+%this \enquote{label-ref system} in various ways.
+
+Over the years \LaTeX{} packages appeared that extended
+this basic \enquote{label-ref system} in various ways.
+%
+For example, the \pkg{refcount} package made a small but significant change
+to the functions used to access recorded values, by making them expandable.
+And the \pkg{smart-ref} package supports the storage of a larger
+collection of counter values so that, for example, a cross-reference
+can refer to the relevant chapter together with an equation tag.
+The \pkg{cleveref} package stores (by means of a second, internal
+\enquote{logical label}) extra information such as the name of the counter.
+The \pkg{hyperref} package adds the \cs{autoref} command, which
+tries to retrieve the name of a counter from the  \emph{logical name}
+used for a link target.
+The \pkg{tikzmarks} package records information about
+ \emph{labelled positions} in pictures when using \pkg{tikz}.
```

Comment:
tikzmark is not restricted to pictures. It can be used everywhere in the text. I wouldn't try to be too exact here and replace "in pictures when using tikz" by "on the page"-

Reply:
good point. but what is also wrong I just noticed is that this is not package but a tikzlibrary

Reply:
OK: the important point is that it is a tikz module rather than a standalone package.

Reply:
Thanks for the correction re “only pictures” :  I
I could not properly understand the documentation since (naturally) it assumed great familiarity with the tikz universe.